main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.FindInFiles/Scope.cs

```diff
@@ -225,9 +227,11 @@ public override IEnumerable<FileProvider> GetFiles (ProgressMonitor monitor, Fil
 				monitor.Log.WriteLine (GettextCatalog.GetString ("Looking in project '{0}'", project.Name));
 				var alreadyVisited = new HashSet<string> ();
 				var conf = project.DefaultConfiguration?.Selector;
-				foreach (ProjectFile file in project.GetSourceFilesAsync (conf).Result.Where (f => filterOptions.NameMatches (f.Name) && File.Exists (f.Name))) {
+				foreach (var file in project.Files) {
 					if ((file.Flags & ProjectItemFlags.Hidden) == ProjectItemFlags.Hidden)
 						continue;
+					if (!filterOptions.NameMatches (file.Name) || !File.Exists (file.Name))
```

Comment:
```suggestion
					if (!filterOptions.NameMatches (file.Name)
```

GetFileIsText below does File.Exists.